diff -up eigen-3.4.1/Eigen/src/Core/util/ConfigureVectorization.h.omv~ eigen-3.4.1/Eigen/src/Core/util/ConfigureVectorization.h
--- eigen-3.4.1/Eigen/src/Core/util/ConfigureVectorization.h.omv~	2025-10-20 18:34:35.071328321 +0200
+++ eigen-3.4.1/Eigen/src/Core/util/ConfigureVectorization.h	2025-10-20 18:34:56.681406752 +0200
@@ -445,7 +445,7 @@
 #define EIGEN_VECTORIZE_FMA
 #endif
 
-#if defined(__F16C__) && !defined(EIGEN_GPUCC) && (!EIGEN_COMP_CLANG_STRICT || EIGEN_COMP_CLANG>=380)
+#if defined(__F16C__) && !defined(EIGEN_GPUCC) && (EIGEN_COMP_CLANG>=380)
   // We can use the optimized fp16 to float and float to fp16 conversion routines
   #define EIGEN_HAS_FP16_C
 
