diff -up eigen-5.0.1/Eigen/src/Core/util/ConfigureVectorization.h.1~ eigen-5.0.1/Eigen/src/Core/util/ConfigureVectorization.h
--- eigen-5.0.1/Eigen/src/Core/util/ConfigureVectorization.h.1~	2025-11-08 21:44:56.000000000 +0100
+++ eigen-5.0.1/Eigen/src/Core/util/ConfigureVectorization.h	2026-03-24 21:44:08.984503828 +0100
@@ -462,7 +462,7 @@ extern "C" {
 #define EIGEN_VECTORIZE_FMA
 #endif
 
-#if defined(__F16C__) && !defined(EIGEN_GPUCC) && (!EIGEN_COMP_CLANG_STRICT || EIGEN_CLANG_STRICT_AT_LEAST(3, 8, 0))
+#if defined(__F16C__) && !defined(EIGEN_GPUCC) && (EIGEN_CLANG_STRICT_AT_LEAST(3, 8, 0))
 // We can use the optimized fp16 to float and float to fp16 conversion routines
 #define EIGEN_HAS_FP16_C
 
